src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoContext } from '../../TodoContext';
+
+interface Props {
+  todo: Todo
+}
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    todos,
+    setTodos,
+    deleteTodo,
+    toggleTodo,
+  } = React.useContext(TodoContext);
+  const [selectedTodo, setSelectedTodo] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const selectedField = React.useRef<HTMLInputElement>(null);
+
+  const handleTitleChange = () => {
+    setSelectedTodo(false);
+    if (newTitle === '') {
+      deleteTodo(todo.id);
+    } else {
+      const newTodos = todos.map(t => ({
+        ...t,
+        title: t.id === todo.id
+          ? newTitle
+          : t.title,
+      }));
+
+      setTodos(newTodos);
+    }
```

Comment:
why do we need to control todos through this common setTodos if we need to change only title? Create `handleChangeTitle` in your context provider, and use it here